File: web-frontend/src/services/quizService.js
import api from './api';

export const quizService = {
  getAll: async () => (await api.get('/quizzes')).data,
  getById: async (id) => (await api.get(`/quizzes/${id}`)).data,
  create: async (data) => (await api.post('/quizzes', data)).data,
  submit: async (id, answers) => (await api.post(`/quizzes/${id}/submit`, answers)).data,
};

File: mobile-app/lib/services/quiz_service.dart
import 'api_service.dart';
class QuizService {
  final ApiService _api = ApiService();
  Future<List<dynamic>> getAll() async => await _api.get('/quizzes');
  Future<dynamic> submit(int quizId, Map<String, dynamic> data) async => await _api.post('/quizzes/$quizId/submit', data);
}
